src/index.ts

```diff
@@ -203,158 +232,217 @@ function getPropertyNames(schema: Schema) {
   return schema.properties.map((item: Property) => item.name).join(', ');
 }
 
-function isEnum(prop: PropertyLike) {
-  return prop['$ref'] && prop['$ref']['enum'];
-}
-
-function isString(param: Parameter) {
-  return !param['$ref'] && param.type === 'string' && !param.format;
-}
-
 function getHandleType(prop: Parameter) {
   if (prop.contentType === 'application/json') {
     return 'char';
-  }
-  else if (isEnum(prop) && prop.contentType === 'text/plain; charset=utf-8') {
+  } else if (helpers.isEnum(prop) && prop.contentType === 'text/plain; charset=utf-8') {
     return 'char';
-  }
-  else if (!prop['$ref']) {
-    if (prop.contentType === "application/x-binary") {
-      if (prop.type === 'buffer') {
-        return 'uint8_t';
-      } else if (prop.type === 'array' && !(prop.items as Property)['$ref']) {
-        // support fixed width numeric types
-        const aprop = prop.items as Property;
-        if (aprop.type === 'number' || aprop.type === 'integer') {
-          return toCppType(aprop, '');
-        }
-      }
-    } else if (prop.contentType === 'text/plain; charset=utf-8') {
-      if (prop.type === 'string' && !prop.format) {
-        return 'char';
-      } else if (prop.type === 'buffer') {
-        return 'char';
+  } else if (prop.contentType === "application/x-binary") {
+    if (helpers.isBuffer(prop)) {
+      return 'uint8_t';
+    } else if (helpers.isArray(prop)) {
```

Comment:
so this would be like:

```yaml
exports:
  foo:
    parameters:
      input:
        contentType: application/x-binary
        type: array
        items:
          type: float
```

Reply:
Almost, I think array items are encoded as type and format so `float` would be represented as:
```yaml
exports:
  foo:
    parameters:
      input:
        contentType: application/x-binary
        type: array
        items:
          type: number
          format: float
```

The schema should be the same as a JSON encoded array of floats, except the content type is binary.